src/App.js

```diff
@@ -35,6 +39,14 @@ function App() {
       return <Navigate to="/login" />;
     }
   }
+  function AdminRouteMiddleware({ children }) {
+    const { isAdmin, checkAdminAuth } = useAuth();
+    useEffect(() => {
```

Comment:
Remove this useEffect. This has no effect here. Call the checkAdminAuth outside this useEffect hook.